### ISSUE TITLE
[Android] Troubleshooting why MediaCodec is not being used (or failing)

### ISSUE THREAD

**Issue by stuckless** [Bilibili/ijkplayer#819]
I have a number of h264 mkvs/mp4s and when I play them in ijkplayer, they play fine, but, I can see from the logging the avcodec is being used instead of MediaCodec (ie software decoding vs hardware decoding).  I'm wondering how I go about determining "why" ijkplayer goes to software decoding or why hardware decoding fails.  These files play in ExoPlayer and it will only play hardware decoded streams, so I know they "should" be able to use the hardware decoder.

Here's my log...

```
01-09 10:29:30.024 4014-4014/sagex.miniclient.android D/IJKMEDIA: ijkmp_set_data_source(url="ijkmediadatasource:1051606")
01-09 10:29:30.024 4014-4014/sagex.miniclient.android D/IJKMEDIA: ijkmp_set_data_source(url="ijkmediadatasource:1051606")=0
01-09 10:29:30.024 4014-4014/sagex.miniclient.android D/IJKMEDIA: IjkMediaPlayer_prepareAsync
01-09 10:29:30.024 4014-4014/sagex.miniclient.android D/IJKMEDIA: ijkmp_prepare_async()
01-09 10:29:30.024 4014-4014/sagex.miniclient.android I/IJKMEDIA: ===== versions =====
01-09 10:29:30.024 4014-4014/sagex.miniclient.android I/IJKMEDIA: FFmpeg       : ff2.8--ijk0.4.1.1--dev0.3.3--rc4
01-09 10:29:30.024 4014-4014/sagex.miniclient.android I/IJKMEDIA: libavutil    : 54.31.100
01-09 10:29:30.024 4014-4014/sagex.miniclient.android I/IJKMEDIA: libavcodec   : 56.60.100
01-09 10:29:30.024 4014-4014/sagex.miniclient.android I/IJKMEDIA: libavformat  : 56.40.101
01-09 10:29:30.024 4014-4014/sagex.miniclient.android I/IJKMEDIA: libswscale   : 3.1.101
01-09 10:29:30.025 4014-4014/sagex.miniclient.android I/IJKMEDIA: libswresample: 1.2.101
01-09 10:29:30.025 4014-4014/sagex.miniclient.android I/IJKMEDIA: ===== options =====
01-09 10:29:30.025 4014-4014/sagex.miniclient.android I/IJKMEDIA: player-opts : mediacodec-avc               = 1
01-09 10:29:30.025 4014-4014/sagex.miniclient.android I/IJKMEDIA: player-opts : mediacodec-hevc              = 1
01-09 10:29:30.025 4014-4014/sagex.miniclient.android I/IJKMEDIA: player-opts : framedrop                    = 1
01-09 10:29:30.025 4014-4014/sagex.miniclient.android I/IJKMEDIA: format-opts : ijkinject-opaque             = 2100262
01-09 10:29:30.025 4014-4014/sagex.miniclient.android I/IJKMEDIA: codec-opts  : skip_loop_filter             = 48
01-09 10:29:30.025 4014-4014/sagex.miniclient.android I/IJKMEDIA: ===================
01-09 10:29:30.025 4014-6509/sagex.miniclient.android I/IJKMEDIA: SDL_RunThread: [6509] ff_msg_loop
01-09 10:29:30.025 4014-6509/sagex.miniclient.android D/IJKMEDIA: message_loop
01-09 10:29:30.026 4014-4014/sagex.miniclient.android D/IJKMEDIA: ijkmp_prepare_async()=0
01-09 10:29:30.026 4014-4014/sagex.miniclient.android D/sagex.miniclient.android.video.ijkplayer.IJKMediaPlayerImpl: [main] DEBUG - mediaplayer has our URL
01-09 10:29:30.027 4014-6510/sagex.miniclient.android I/IJKMEDIA: SDL_RunThread: [6510] ff_vout
01-09 10:29:30.027 4014-6509/sagex.miniclient.android D/IJKMEDIA: FFP_MSG_FLUSH:
01-09 10:29:30.027 4014-6511/sagex.miniclient.android I/IJKMEDIA: SDL_RunThread: [6511] ff_read
01-09 10:29:30.028 4014-4876/sagex.miniclient.android D/sagex.miniclient.android.video.ijkplayer.IJKMediaPlayerImpl: [Media-192.168.1.176] DEBUG - play()
01-09 10:29:30.028 4014-4876/sagex.miniclient.android D/IJKMEDIA: IjkMediaPlayer_start
01-09 10:29:30.028 4014-4876/sagex.miniclient.android D/IJKMEDIA: ijkmp_start()
01-09 10:29:30.028 4014-4876/sagex.miniclient.android D/IJKMEDIA: ijkmp_start()=-3
01-09 10:29:30.044 4014-6511/sagex.miniclient.android E/IJKMEDIA: Option ijkinject-opaque not found.
01-09 10:29:30.069 4014-6511/sagex.miniclient.android I/IJKMEDIA: max_frame_duration: 10.000
01-09 10:29:30.069 4014-6511/sagex.miniclient.android I/IJKMEDIA: Input #0, matroska,webm, from 'ijkmediadatasource:1051606':
01-09 10:29:30.069 4014-6511/sagex.miniclient.android I/IJKMEDIA:   Metadata:
01-09 10:29:30.069 4014-6511/sagex.miniclient.android I/IJKMEDIA:     encoder         : 
01-09 10:29:30.069 4014-6511/sagex.miniclient.android I/IJKMEDIA: libebml v1.3.0 + libmatroska v1.4.1
01-09 10:29:30.069 4014-6511/sagex.miniclient.android I/IJKMEDIA:     creation_time   : 
01-09 10:29:30.069 4014-6511/sagex.miniclient.android I/IJKMEDIA: 2015-10-12 03:15:16
01-09 10:29:30.069 4014-6511/sagex.miniclient.android I/IJKMEDIA:   Duration: 
01-09 10:29:30.069 4014-6511/sagex.miniclient.android I/IJKMEDIA: 01:04:57.41
01-09 10:29:30.069 4014-6511/sagex.miniclient.android I/IJKMEDIA: , start: 
01-09 10:29:30.069 4014-6511/sagex.miniclient.android I/IJKMEDIA: 0.000000
01-09 10:29:30.069 4014-6511/sagex.miniclient.android I/IJKMEDIA: , bitrate: 
01-09 10:29:30.069 4014-6511/sagex.miniclient.android I/IJKMEDIA: 7448 kb/s
01-09 10:29:30.069 4014-6511/sagex.miniclient.android I/IJKMEDIA:     Stream #0:0
01-09 10:29:30.069 4014-6511/sagex.miniclient.android I/IJKMEDIA: (eng)
01-09 10:29:30.069 4014-6511/sagex.miniclient.android I/IJKMEDIA: : Video: h264 (High), yuv420p(tv, bt709/unknown/unknown), 1280x720
01-09 10:29:30.069 4014-6511/sagex.miniclient.android I/IJKMEDIA: , SAR 1:1 DAR 16:9
01-09 10:29:30.069 4014-6511/sagex.miniclient.android I/IJKMEDIA: , 
01-09 10:29:30.069 4014-6511/sagex.miniclient.android I/IJKMEDIA: 23.98 fps, 
01-09 10:29:30.069 4014-6511/sagex.miniclient.android I/IJKMEDIA: 23.98 tbr, 
01-09 10:29:30.069 4014-6511/sagex.miniclient.android I/IJKMEDIA: 1k tbn, 
01-09 10:29:30.069 4014-6511/sagex.miniclient.android I/IJKMEDIA: 47.95 tbc
01-09 10:29:30.069 4014-6511/sagex.miniclient.android I/IJKMEDIA:  (default)
01-09 10:29:30.069 4014-6511/sagex.miniclient.android I/IJKMEDIA:     Stream #0:1
01-09 10:29:30.069 4014-6511/sagex.miniclient.android I/IJKMEDIA: : Audio: ac3, 48000 Hz, 5.1(side), fltp, 384 kb/s
01-09 10:29:30.069 4014-6511/sagex.miniclient.android I/IJKMEDIA:  (default)
01-09 10:29:30.070 4014-6511/sagex.miniclient.android E/IJKMEDIA: SDL_Android_AudioTrack_new_from_spec: invalid channel 0
01-09 10:29:30.070 4014-6511/sagex.miniclient.android E/IJKMEDIA: aout_open_audio_n: failed to new AudioTrcak()
01-09 10:29:30.070 4014-6511/sagex.miniclient.android W/IJKMEDIA: SDL_OpenAudio (6 channels, 48000 Hz): (null)
01-09 10:29:30.070 4014-6511/sagex.miniclient.android E/IJKMEDIA: SDL_Android_AudioTrack_new_from_spec: invalid channel 0
01-09 10:29:30.070 4014-6511/sagex.miniclient.android E/IJKMEDIA: aout_open_audio_n: failed to new AudioTrcak()
01-09 10:29:30.070 4014-6511/sagex.miniclient.android W/IJKMEDIA: SDL_OpenAudio (4 channels, 48000 Hz): (null)
01-09 10:29:30.070 4014-6511/sagex.miniclient.android I/IJKMEDIA: SDL_Android_AudioTrack: CHANNEL_OUT_STEREO
01-09 10:29:30.070 4014-6511/sagex.miniclient.android I/IJKMEDIA: SDL_Android_AudioTrack: ENCODING_PCM_16BIT
01-09 10:29:30.073 4014-6511/sagex.miniclient.android I/IJKMEDIA: SDL_Android_AudioTrack_new_from_spec: init volume as 1.000000/(1.000000,0.000000)
01-09 10:29:30.073 4014-6511/sagex.miniclient.android I/IJKMEDIA: audio_session_id = 226
01-09 10:29:30.073 4014-6511/sagex.miniclient.android I/IJKMEDIA: AudioCodec: avcodec, ac3
01-09 10:29:30.073 4014-6518/sagex.miniclient.android I/IJKMEDIA: SDL_RunThread: [6518] ff_audio_dec
01-09 10:29:30.073 4014-6517/sagex.miniclient.android I/IJKMEDIA: SDL_RunThread: [6517] ff_aout_android
01-09 10:29:30.076 4014-6511/sagex.miniclient.android D/IJKMEDIA: ffpipenode_create_video_decoder_from_android_mediacodec()
01-09 10:29:30.076 4014-6511/sagex.miniclient.android I/IJKMEDIA: ffpipenode_create_video_decoder_from_android_mediacodec: MediaCodec: H264_HIGH: enabled
01-09 10:29:30.076 4014-6511/sagex.miniclient.android I/IJKMEDIA: AMediaFormat: video/avc, 1280x720
01-09 10:29:30.076 4014-6511/sagex.miniclient.android D/IJKMEDIA: SDL_AMediaFormatJava_createVideoFormat
01-09 10:29:30.076 4014-6511/sagex.miniclient.android E/IJKMEDIA: csd-0[40]: 00000001
01-09 10:29:30.076 4014-6511/sagex.miniclient.android E/IJKMEDIA: csd-0[40]: 67640029
01-09 10:29:30.076 4014-6511/sagex.miniclient.android E/IJKMEDIA: csd-0[40]: acd98050
01-09 10:29:30.076 4014-6511/sagex.miniclient.android E/IJKMEDIA: csd-0[40]: 05ba6a04
01-09 10:29:30.076 4014-6511/sagex.miniclient.android E/IJKMEDIA: csd-0[40]: 04028000
01-09 10:29:30.076 4014-6511/sagex.miniclient.android E/IJKMEDIA: csd-0[40]: 01f48000
01-09 10:29:30.076 4014-6511/sagex.miniclient.android E/IJKMEDIA: csd-0[40]: 5dc0478c
01-09 10:29:30.076 4014-6511/sagex.miniclient.android E/IJKMEDIA: csd-0[40]: 18cd0000
01-09 10:29:30.076 4014-6511/sagex.miniclient.android E/IJKMEDIA: csd-0[40]: 000168e9
01-09 10:29:30.076 4014-6511/sagex.miniclient.android E/IJKMEDIA: csd-0[40]: 78232c8b
01-09 10:29:30.076 4014-6511/sagex.miniclient.android I/IJKMEDIA: amc: rotate notify: 0
01-09 10:29:30.076 4014-6511/sagex.miniclient.android D/IJKMEDIA: ffpipeline_select_mediacodec_l
01-09 10:29:30.076 4014-6509/sagex.miniclient.android D/IJKMEDIA: FFP_MSG_VIDEO_ROTATION_CHANGED: 0
01-09 10:29:30.076 4014-6511/sagex.miniclient.android I/tv.danmaku.ijk.media.player.IjkMediaPlayer: onSelectCodec: mime=video/avc, profile=100, level=41
01-09 10:29:30.076 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:   found codec: OMX.Nvidia.mp4.decode
01-09 10:29:30.076 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:     mime: video/mp4v-es
01-09 10:29:30.076 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:   found codec: OMX.Nvidia.h263.decode
01-09 10:29:30.076 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:     mime: video/3gpp
01-09 10:29:30.076 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:   found codec: OMX.Nvidia.h264.decode
01-09 10:29:30.076 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:     mime: video/avc
01-09 10:29:30.076 4014-6511/sagex.miniclient.android I/tv.danmaku.ijk.media.player.IjkMediaPlayer: candidate codec: OMX.Nvidia.h264.decode rank=800
01-09 10:29:30.077 4014-6511/sagex.miniclient.android I/IjkMediaCodecInfo:  High Profile Level 51 (8,32768)
01-09 10:29:30.077 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:   found codec: OMX.Nvidia.vp8.decode
01-09 10:29:30.077 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:     mime: video/x-vnd.on2.vp8
01-09 10:29:30.077 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:   found codec: OMX.Nvidia.vp9.decode
01-09 10:29:30.077 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:     mime: video/x-vnd.on2.vp9
01-09 10:29:30.077 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:   found codec: OMX.Nvidia.h265.decode
01-09 10:29:30.077 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:     mime: video/hevc
01-09 10:29:30.077 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:   found codec: OMX.Nvidia.mpeg2v.decode
01-09 10:29:30.077 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:     mime: video/mpeg2
01-09 10:29:30.077 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:   found codec: OMX.Nvidia.vc1.decode
01-09 10:29:30.077 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:     mime: video/wvc1
01-09 10:29:30.077 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:     mime: video/x-ms-wmv
01-09 10:29:30.077 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:   found codec: OMX.Nvidia.mp3.decoder
01-09 10:29:30.077 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:     mime: audio/mpeg
01-09 10:29:30.077 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:   found codec: OMX.Nvidia.mp2.decoder
01-09 10:29:30.077 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:     mime: audio/mpeg-L2
01-09 10:29:30.077 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:   found codec: OMX.Nvidia.eaacp.decoder
01-09 10:29:30.077 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:     mime: audio/mp4a-latm
01-09 10:29:30.077 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:   found codec: OMX.Nvidia.wma.decoder
01-09 10:29:30.077 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:     mime: audio/x-ms-wma
01-09 10:29:30.077 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:   found codec: OMX.Nvidia.mjpeg.decoder
01-09 10:29:30.077 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:     mime: video/mjpeg
01-09 10:29:30.077 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:   found codec: OMX.Nvidia.h264.encoder
01-09 10:29:30.077 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:   found codec: OMX.Nvidia.vp8.encoder
01-09 10:29:30.077 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:   found codec: OMX.google.mp3.decoder
01-09 10:29:30.077 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:     mime: audio/mpeg
01-09 10:29:30.077 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:   found codec: OMX.google.amrnb.decoder
01-09 10:29:30.077 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:     mime: audio/3gpp
01-09 10:29:30.077 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:   found codec: OMX.google.amrwb.decoder
01-09 10:29:30.077 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:     mime: audio/amr-wb
01-09 10:29:30.077 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:   found codec: OMX.google.aac.decoder
01-09 10:29:30.077 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:     mime: audio/mp4a-latm
01-09 10:29:30.077 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:   found codec: OMX.google.g711.alaw.decoder
01-09 10:29:30.077 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:     mime: audio/g711-alaw
01-09 10:29:30.077 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:   found codec: OMX.google.g711.mlaw.decoder
01-09 10:29:30.077 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:     mime: audio/g711-mlaw
01-09 10:29:30.077 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:   found codec: OMX.google.vorbis.decoder
01-09 10:29:30.077 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:     mime: audio/vorbis
01-09 10:29:30.077 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:   found codec: OMX.google.opus.decoder
01-09 10:29:30.077 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:     mime: audio/opus
01-09 10:29:30.077 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:   found codec: OMX.google.raw.decoder
01-09 10:29:30.077 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:     mime: audio/raw
01-09 10:29:30.077 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:   found codec: OMX.google.aac.encoder
01-09 10:29:30.077 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:   found codec: OMX.google.amrnb.encoder
01-09 10:29:30.077 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:   found codec: OMX.google.amrwb.encoder
01-09 10:29:30.077 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:   found codec: OMX.google.flac.encoder
01-09 10:29:30.077 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:   found codec: OMX.google.mpeg4.decoder
01-09 10:29:30.077 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:     mime: video/mp4v-es
01-09 10:29:30.077 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:   found codec: OMX.google.h263.decoder
01-09 10:29:30.077 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:     mime: video/3gpp
01-09 10:29:30.077 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:   found codec: OMX.google.h264.decoder
01-09 10:29:30.077 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:     mime: video/avc
01-09 10:29:30.077 4014-6511/sagex.miniclient.android I/tv.danmaku.ijk.media.player.IjkMediaPlayer: candidate codec: OMX.google.h264.decoder rank=200
01-09 10:29:30.078 4014-6511/sagex.miniclient.android I/IjkMediaCodecInfo:  Baseline Profile Level 51 (1,32768)
01-09 10:29:30.078 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:   found codec: OMX.google.hevc.decoder
01-09 10:29:30.078 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:     mime: video/hevc
01-09 10:29:30.078 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:   found codec: OMX.google.vp8.decoder
01-09 10:29:30.078 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:     mime: video/x-vnd.on2.vp8
01-09 10:29:30.078 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:   found codec: OMX.google.vp9.decoder
01-09 10:29:30.078 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:     mime: video/x-vnd.on2.vp9
01-09 10:29:30.078 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:   found codec: OMX.google.h263.encoder
01-09 10:29:30.078 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:   found codec: OMX.google.h264.encoder
01-09 10:29:30.078 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:   found codec: OMX.google.mpeg4.encoder
01-09 10:29:30.078 4014-6511/sagex.miniclient.android D/tv.danmaku.ijk.media.player.IjkMediaPlayer:   found codec: OMX.google.vp8.encoder
01-09 10:29:30.078 4014-6511/sagex.miniclient.android I/tv.danmaku.ijk.media.player.IjkMediaPlayer: selected codec: OMX.Nvidia.h264.decode rank=800
01-09 10:29:30.078 4014-6511/sagex.miniclient.android D/IJKMEDIA: SDL_AMediaCodecJava_createByCodecName
01-09 10:29:30.082 4014-6511/sagex.miniclient.android D/IJKMEDIA: SDL_AMediaCodecJava_init
01-09 10:29:30.082 4014-6511/sagex.miniclient.android D/IJKMEDIA: SDL_AMediaCodec_increaseReference(): ref=1
01-09 10:29:30.082 4014-6511/sagex.miniclient.android D/IJKMEDIA: SDL_AMediaCodecJava_configure_surface
01-09 10:29:30.082 4014-6511/sagex.miniclient.android E/IJKMEDIA: configure acodec:0x100c4e format:0xc32: surface:0x100c3e
01-09 10:29:30.085 4014-6511/sagex.miniclient.android E/IJKMEDIA: reconfigure_codec_l:configure_surface: failed
01-09 10:29:30.085 4014-6511/sagex.miniclient.android D/IJKMEDIA: SDL_AMediaCodec_decreaseReference(): ref=0
01-09 10:29:30.085 4014-6511/sagex.miniclient.android I/IJKMEDIA: SDL_AMediaCodecJava_delete
01-09 10:29:30.088 4014-6511/sagex.miniclient.android I/IJKMEDIA: VideoCodec: avcodec, h264
01-09 10:29:30.088 4014-6511/sagex.miniclient.android W/IJKMEDIA: fps: 23.976024 (normal)
01-09 10:29:30.088 4014-6511/sagex.miniclient.android W/IJKMEDIA: fps: 23.976024 (normal)

```

**Comment by stuckless** [comment]
Something that I just noticed.  The first time I play a video, it's fine.  In fact it's fine everytime, until, I play a video that requires software decoding (ie, mpeg2 video) and then every video I play after that, uses the software decoder, even it played it fine a few seconds before.

I release the player every time.

I don't however create a new SurfaceView each time I play a video.  I re-use the previous SurfaceView.

**Comment by bbcallen** [comment]
However, It's very likely that previous player instance or MediaCodec instance is not released properly.
Can you paste the log about the last HW player and the first SW player?

**Comment by stuckless** [comment]
I'll try to get that log.  I've actually worked around this issue after finding an obscure reference to a person having a similar issue with the Android MediaPlayer.  They fixed it by setting the SurfaceView visibility to GONE when the player is released, and then again to VISIBLE when the player is being recreated.

I have no idea why setting the visiblity to GONE/VISIBLE would fix this unless the SurfaceView has some reinitialization code in there??

So right now, using the GONE/VISIBLE hack, it all works.  Setting the SurfaceView to GONE was something I needed to do anyways, since without that, I would get the last frame of the player still visible on the view and it would bleed through my UI.

I'm OK if you want to close this issue, and I can revisit it, if it turns up again.
